File: project/design_pattern/simple_factory/operation_factory.cpp
#include "operation_factory.h"

Operation *OperationFactory::createOperation(string operation)
{
	Operation *op = NULL;

	if(operation == "+")
	{
		op = new OperationAdd();
	}
	else if(operation == "-")
	{
		op = new OperationSub();
	}
	else
	{
		return NULL;
	}
	return op;
}

File: project/JavaDesignPattern/src/yue/soft/design/pattern/samplefactory/OperationFactory.java
package yue.soft.design.pattern.samplefactory;

public class OperationFactory {
	public static Operation createOperation(String operation){
		Operation oper = null;
		if(operation.equals("+")){
			oper = new OperationAdd();
		}else if(operation.equals("-")){
			oper = new OperationSub();
		}else{
			return null;
		}
		return oper;
	}
}
